module/Frontpage/src/Frontpage/Service/Page.php

```diff
@@ -0,0 +1,276 @@
+                    // Page url is /$category
+                    $pageArray[$category]['page'] = $page;
+                } else {
+                    $pageArray[$category]['children'][$subCategory]['page'] = $page;
+                    // Page url is /$category/$subCategory
+                }
+            } else {
+                // Page url is /$category/$subCategory/$name
+                $pageArray[$category]['children'][$subCategory]['children'][$name]['page'] = $page;
+            }
+        }
+
+        return $pageArray;
+    }
+
+    /**
+     * Creates a new Page.
+     *
+     * @param array $data form post data
+     * @return bool|PageModel false if creation was not successful.
+     */
+    public function createPage($data)
+    {
+        $form = $this->getPageForm();
+        $page = new PageModel();
+        $form->bind($page);
+        $form->setData($data);
+
+        if (!$form->isValid()) {
+            return false;
+        }
+
+        $this->getPageMapper()->persist($page);
+        $this->getPageMapper()->flush();
+
+        return $page;
+    }
+
+    /**
+     * @param integer $pageId
+     * @param array $data form post data
+     * @return bool
+     */
+    public function updatePage($pageId, $data)
+    {
+        if (!$this->isAllowed('edit')) {
+            throw new \User\Permissions\NotAllowedException(
+                $this->getTranslator()->translate('You are not allowed to edit pages.')
+            );
+        }
+        $form = $this->getPageForm($pageId);
+        $form->setData($data);
+
+        if (!$form->isValid()) {
+            return false;
+        }
+
+        $this->getPageMapper()->flush();
+
+        return true;
+    }
+
+    /**
+     * Removes a page.
+     *
+     * @param integer $pageId The id of the page to remove.
+     */
+    public function deletePage($pageId)
+    {
+        $page = $this->getPageById($pageId);
+        $this->getPageMapper()->remove($page);
+        $this->getPageMapper()->flush();
+    }
+
+    /**
+     * Upload an image to be displayed on a page.
+     *
+     * @param array $files
+     *
+     * @throws \Exception
+     * @return array
+     */
+    public function uploadImage($files)
+    {
+        $imageValidator = new \Zend\Validator\File\IsImage(
+            array('magicFile' => false)
+        );
+
+        $extensionValidator = new \Zend\Validator\File\Extension(
+            array('JPEG', 'JPG', 'JFIF', 'TIFF', 'RIF', 'GIF', 'BMP', 'PNG')
+        );
+
+        $translator = $this->getTranslator();
+
+        if ($files['upload']['error'] !== 0) {
+            throw new \Exception(
+                $translator->translate('An unknown error occurred during uploading (' . $files['upload']['error'] . ')')
+            );
+        }
+
+        if ($imageValidator->isValid($files['upload']['tmp_name'])) {
+
+
+            if ($extensionValidator->isValid($files['upload']['name'])) {
+
+                //TODO
```

Comment:
Deze kan pas opgelost worden zodra de file storage service af is, waar we het eerder over hadden gehad. Dus dit komt in een laters PR goed.